File: ARRAY/2D_Array/TwoD_Array.py
# Asking the user for number of rows and columns
rows = int(input("Enter the number of rows: "))
cols = int(input("Enter the number of columns: "))

# Calculate total elements
total_elements = rows * cols
print(f"You can enter a total of {total_elements} elements.")

# Create the 2D array
matrix = []

# Taking user input for elements
print("Enter the elements of the matrix:")
for i in range(rows):
    row = list(map(int, input().split()))  # Taking space-separated input for each row
    matrix.append(row)

# Printing the matrix
print("The entered matrix is:")
for row in matrix:
    print(" ".join(map(str, row)))

File: ARRAY/2D_Array/TwoD_Array.c
#include <stdio.h>

int main() {
    int rows, cols;

    // Ask user for number of rows and columns
    printf("Enter the number of rows: ");
    scanf("%d", &rows);
    printf("Enter the number of columns: ");
    scanf("%d", &cols);

    // Calculate total elements
    int totalElements = rows * cols;
    printf("You can enter a total of %d elements.\n", totalElements);

    // Create the 2D array
    int arr[rows][cols];

    // Taking user input for elements
    printf("Enter the elements of the matrix:\n");
    for (int i = 0; i < rows; i++) {
        for (int j = 0; j < cols; j++) {
            scanf("%d", &arr[i][j]);
        }
    }

    // Printing the matrix
    printf("The entered matrix is:\n");
    for (int i = 0; i < rows; i++) {
        for (int j = 0; j < cols; j++) {
            printf("%d ", arr[i][j]);
        }
        printf("\n");
    }

    return 0;
}
